### PR TITLE
After deploy modal with Twitter share and project payer deploy

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

**Note: ** this targets #901 , so let's get that one reviewed and in first.

Adds a new post-deploy modal:
1. Issue token
2. Deploy payable address
3.  Share to Twitter

Test by creating a new V2 project and going through the steps if you please :)

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/166170198-9b2c8652-b937-41e3-b2b9-014391940952.mp4

New modal with `RichButton`s:

Screen Shot 2022-05-03 at 5 01 45 pm

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Comment by jmilldotdev** [comment]
LGTM

File: src/components/shared/modals/IssueTokenModal.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    const fields = form.getFieldsValue(true)
+
+    issueTokensTx(
+      { name: fields.name, symbol: fields.symbol },
+      {
+        onDone: () => {
+          setTransactionPending(true)
+        },
+        onConfirmed: () => {
```
Comment (benschac) [inline src/components/shared/modals/IssueTokenModal.tsx:40]: I know this is v2. Assume we could add similar `onError` logic that's in v1.

**Comment by benschac** [comment]
I didn't have time to go too far in-depth or leave an LGTM. But it looks good so far!

File: src/components/v2/V2Project/NewDeployModal.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      okText={t`Done`}
+      cancelText={t`Later`}
+    >
+      <h2>
+        <Trans>Next steps</Trans>
+      </h2>
+      <p>
+        <Trans>
+          These are all optional and can be completed at any time through your
+          project page.
```
Comment (aeolianeth) [inline src/components/v2/V2Project/NewDeployModal.tsx:64]:
```suggestion
          Congratulations on launching your project! We recommend you take these next steps to help your project gain traction. These steps are optional and you can complete them later on.
```

File: src/components/v2/V2Project/NewDeployModal.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          primaryColor={hasIssuedToken ? seenColor : undefined}
+          style={stepButtonStyle}
+        />
+        <RichButton
+          prefix="2"
+          heading={<Trans>Create a payable address</Trans>}
+          description={
+            <Trans>
+              Create an ETH address people can use to pay this project rather
+              than paying through the juicebox.money interface.
```
Comment (aeolianeth) [inline src/components/v2/V2Project/NewDeployModal.tsx:93]:
```suggestion
              Create an Ethereum address that can be used to pay your project directly.
```

File: src/components/v2/V2Project/NewDeployModal.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  visible,
+  onClose,
+}: {
+  visible: boolean
+  onClose: VoidFunction
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const projectURL = window.location.origin + useLocation().pathname
```
Comment (aeolianeth) [inline src/components/v2/V2Project/NewDeployModal.tsx:26]: is `origin` returned in the `useLocation()` hook?

File: src/components/v2/V2Project/NewDeployModal.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+              <CheckCircleFilled style={{ color: seenColor }} />
+            ) : undefined
+          }
+          primaryColor={hasLaunchedPayableAddress ? seenColor : undefined}
+          style={stepButtonStyle}
+        />
+        <RichButton
+          prefix="3"
+          heading={<Trans>Share to Twitter</Trans>}
+          description={<Trans>Let's get this party started!</Trans>}
```
Comment (aeolianeth) [inline src/components/v2/V2Project/NewDeployModal.tsx:109]: Love

**Review by aeolianeth — CHANGES_REQUESTED** [review]
Coupla comments, assign back to me for merge when ready 🙏

File: src/components/v2/V2Project/NewDeployModal.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NewDeployModal.tsx:26]: Nah it's not

**Review by aeolianeth — APPROVED** [review]
@johnnyd-eth LGTM

**Pull request merged**